### PR TITLE
[AMS-426] add Prometheus metrics

### PR DESCRIPTION
**Pull request by xakep139**

### CODE REVIEW

File: src/VStore.Host/Controllers/DevController.cs [hunk 1]
```diff
@@ -1,49 +0,0 @@
...
-using Microsoft.Extensions.Options;
-
-using NuClear.VStore.Locks;
-using NuClear.VStore.Options;
-
-namespace NuClear.VStore.Host.Controllers
-{
-    [ApiVersionNeutral]
-    [Route("api/{api-version:apiVersion}/dev")]
-    public sealed class DevController : VStoreController
```
Comment (xakep139) [inline src/VStore.Host/Controllers/DevController.cs]: Remove `dev` controller

File: src/VStore/Templates/TemplatesStorageReader.cs [hunk 2]
```diff
@@ -133,13 +134,14 @@ public TemplatesStorageReader(CephOptions cephOptions, IAmazonS3 amazonS3)
 
         public async Task<bool> IsTemplateExists(long id)
         {
+            var strId = id.ToString();
```
Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Call `ToString()` only once

File: src/VStore/Templates/TemplatesStorageReader.cs [hunk 3]
```diff
@@ -84,45 +84,46 @@ public TemplatesStorageReader(CephOptions cephOptions, IAmazonS3 amazonS3)
...
-            return descriptor;
         }
 
         public async Task<string> GetTemplateLatestVersion(long id)
         {
-            var versionsResponse = await _amazonS3.ListVersionsAsync(_bucketName, id.ToString());
-            var version = versionsResponse.Versions.Find(x => !x.IsDeleteMarker && x.IsLatest);
+            var strId = id.ToString();
+            var versionsResponse = await _amazonS3.ListVersionsAsync(_bucketName, strId);
+            var version = versionsResponse.Versions.Find(x => x.Key == strId && !x.IsDeleteMarker && x.IsLatest);
```
Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Fix bug with ambigious prefixes

File: src/VStore/Templates/TemplatesStorageReader.cs [hunk 4]
```diff
@@ -133,13 +134,14 @@ public TemplatesStorageReader(CephOptions cephOptions, IAmazonS3 amazonS3)
...
-                                           BucketName = _bucketName,
-                                           Prefix = id.ToString()
-                                       });
-            return listResponse.S3Objects.SingleOrDefault(o => o.Key == id.ToString()) != null;
+                                    new ListObjectsV2Request
+                                        {
+                                            BucketName = _bucketName,
+                                            Prefix = strId
+                                    });
+            return listResponse.S3Objects.Any(o => o.Key == strId);
```
Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Small optimization

File: src/VStore/Templates/TemplatesStorageReader.cs (hunk 3 above)
Comment (denisivan0v) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Got it

File: src/VStore/Templates/TemplatesStorageReader.cs (hunk 2 above)
Comment (denisivan0v) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Rename to `idAsString` or simiral

File: src/VStore/Templates/TemplatesStorageReader.cs (hunk 4 above)
Comment (denisivan0v) [inline src/VStore/Templates/TemplatesStorageReader.cs]: `SingleOrDefault` throws an exception if there are more than one instance in sequence, `Any` doesn't

Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: AFAIK, we can't have two objects with the same key, so enumerating whole collection is pointless

File: src/VStore/Templates/TemplatesStorageReader.cs (hunk 2 above)
Comment (xakep139) [inline src/VStore/Templates/TemplatesStorageReader.cs]: Done in https://github.com/2gis/nuclear-vstore/pull/75/commits/a6fb304014302ae9b00dba1418edcd82e2cd25f1

**Pull request merged**